File: backend/src/main/java/de/learneasy/datatransferobjects/ExerciseTemplatesPageDTO.java
package de.learneasy.datatransferobjects;

import de.learneasy.datatransferobjects.exercises.exercisetemplates.ExerciseTemplatePartialDTO;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.Size;
import jakarta.validation.constraints.NotNull;

import java.util.List;

public record ExerciseTemplatesPageDTO(
    @Min(0) @NotNull Integer maximumPage,
    @Size(min = 10, max = 100) @NotNull List<@NotNull ExerciseTemplatePartialDTO> exerciseTemplates
){}

File: frontend/src/data/server/dataTransferObjects/ExerciseTemplatesPageDTO.ts
import type {ExerciseTemplatePartialDTO} from "@/data/server/dataTransferObjects/exercises/exerciseTemplates/ExerciseTemplatePartialDTO.ts";

export type ExerciseTemplatesPageDTO = {
    readonly maximumPage: number
    readonly exerciseTemplates: ReadonlyArray<ExerciseTemplatePartialDTO>
}